### PR TITLE
Add documentation for git hosting

### PR DESCRIPTION
**Pull request by simonepelosi**

### CODE REVIEW

File: reference/services/git-hosting.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+The outside world connects to the Git hosting service using one of several protocols:
+
+* ``git://``
+* ``git+ssh://``
+* ``https://``
+
+The Git "dumb" HTTP transport will not be supported, as it has largely been considered 
+deprecated since the provision of "smart" HTTP transport in Git v1.6.6 (released January, 2010).
+
+SSL is terminated by haproxy for HTTPS, and all the other protocols also pass through
```
Comment (lgp171188) [inline reference/services/git-hosting.rst:34]: Can we mention which unit this haproxy is?

File: reference/services/git-hosting.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+SSL is terminated by haproxy for HTTPS, and all the other protocols also pass through 
+haproxy for load balancing (although at the moment we only have a single backend).  
+The underlying protocol endpoints live in `lp:turnip <https://code.launchpad.net/turnip>`_, 
+which invokes ``git upload-pack`` and ``git receive-pack`` to implement the git protocol itself.
+
+``turnip`` also provides an internal API used by the Launchpad web application to manipulate and inspect repositories; 
+and in turn the Launchpad web application provides an internal XML-RPC interface used by ``turnip`` to translate logical 
+repository paths to filesystem paths and to notify Launchpad of repository changes.
+
+``turnip`` is deployed using Juju, using the `turnip charm <https://git.launchpad.net/turnip/tree/charm>`_.
```
Comment (lgp171188) [inline reference/services/git-hosting.rst:43]:
```suggestion
``turnip`` is deployed using Juju, using the `turnip charms <https://git.launchpad.net/turnip/tree/charm>`_.  
```

File: reference/services/git-hosting.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+The underlying protocol endpoints live in `lp:turnip <https://code.launchpad.net/turnip>`_, 
+which invokes ``git upload-pack`` and ``git receive-pack`` to implement the git protocol itself.
+
+``turnip`` also provides an internal API used by the Launchpad web application to manipulate and inspect repositories; 
+and in turn the Launchpad web application provides an internal XML-RPC interface used by ``turnip`` to translate logical 
+repository paths to filesystem paths and to notify Launchpad of repository changes.
+
+``turnip`` is deployed using Juju, using the `turnip charm <https://git.launchpad.net/turnip/tree/charm>`_.  
+
+See more at `turnip documentation <https://turnip.readthedocs.io/en/latest/index.html>`_
```
Comment (lgp171188) [inline reference/services/git-hosting.rst:45]:
```suggestion
For more details, see the `turnip documentation <https://turnip.readthedocs.io/en/latest/index.html>`_ 
```

File: reference/services/git-hosting.rst (hunk 1 above)
Comment (simonepelosi) [reply reference/services/git-hosting.rst:34]: I think it makes more sense to add a link to our mojo-specs repo to understand how haproxy is configured to handle all the protocols that we support, what do you think ?

File: reference/services/git-hosting.rst [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+* ``rless turnip-pack-{1,2,3,4}.lp.internal::turnip-logs/turnip-access.log``
+* ``rless turnip-pack-{1,2,3,4}.lp.internal::turnip-logs/turnip-api-access.log``
+* ``rless turnip-pack-{1,2,3,4}.lp.internal::turnip-logs/turnip-api-error.log``
+* ``rless turnip-pack-{1,2,3,4}.lp.internal::turnip-logs/turnip-pack-backend.log``
+* ``rless turnip-pack-{1,2,3,4}.lp.internal::turnip-logs/turnip-pack-frontend-git.log``
+* ``rless turnip-pack-{1,2,3,4}.lp.internal::turnip-logs/turnip-pack-frontend-http.log``
+* ``rless turnip-pack-{1,2,3,4}.lp.internal::turnip-logs/turnip-pack-frontend-ssh.log``
+* ``rless turnip-pack-{1,2,3,4}.lp.internal::turnip-logs/turnip-pack-virt.log``
+
+Staging
```
Comment (ineesalmeida) [inline reference/services/git-hosting.rst:80]: *Qastaging?

File: reference/services/git-hosting.rst [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+The web application
+-------------------
+
+Code that is executed as part of the Launchpad web application.  Major features:
+
+ * general information
+ * listings for various registry objects - people, teams, projects, packages
+ * default repositories for projects and packages
+ * privacy
```
Comment (ineesalmeida) [inline reference/services/git-hosting.rst:55]: Perhaps would be clearer as `privacy settings`

File: reference/services/git-hosting.rst (hunk 1 above)
Comment (simonepelosi) [reply reference/services/git-hosting.rst]: Done!

File: reference/services/git-hosting.rst [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+===============
+
+Description
+-----------
+
+Launchpad has direct Git hosting support alongside its existing Bazaar hosting support. 
+The major subsystems involved are:
+
+* The git client
+* Hosting service
```
[the remark this answers is not in the document]
Comment (ineesalmeida) [reply reference/services/git-hosting.rst]: Any thoughts on removing the `The underlying object model` from this list? Don't mind leaving it, but I'd remove it IMO

File: reference/services/git-hosting.rst [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+* ``git://``
+* ``git+ssh://``
+* ``https://``
+
+The Git "dumb" HTTP transport will not be supported, as it has largely been considered 
+deprecated since the provision of "smart" HTTP transport in Git v1.6.6 (released January, 2010).
+
+SSL is terminated by haproxy (haproxy has two units haproxy/1* and haproxy/0) for HTTPS, and all the other 
+protocols also pass through haproxy for load balancing (although at the moment we only have a single backend).  
+See the full charm configuration at `launchpad-mojo-specs <https://code.launchpad.net/launchpad-mojo-specs>_`.
```
Comment (ineesalmeida) [inline reference/services/git-hosting.rst:36]: Very minor, just noticed that the _ at the end of the link should be outside of the ` :)
Also, perhaps we could link to the bundle directly maybe?
https://git.launchpad.net/launchpad-mojo-specs/tree/lp-git/bundle.yaml

File: reference/services/git-hosting.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (simonepelosi) [reply reference/services/git-hosting.rst]: Yep, I removed it to avoid confusion :)

**Review by ineesalmeida — APPROVED** [review]
LGTM! Thank you for adding this great piece of information to our docs!

**Pull request merged**